### DISCUSSION TITLE
Can I pass different image input for multi ControlNet via API?

### DISCUSSION THREAD

**Discussion by OedoSoldier** [Mikubill/sd-webui-controlnet#575] [Q&A] [answered]
While writing a script, it seems that the image passed via `control_net_input_image` is used for all the ControlNets together. Is there any way to pass different image inputs for each ControlNet while generating a single image? For instance, can I pass an OpenPose image to one ControlNet and a segmented image to another ControlNet?

**Comment by ljleb** [comment]
For the web API you can check the documentation: https://github.com/Mikubill/sd-webui-controlnet/wiki/API#web-api

Root `control_net_*` json properties are deprecated, you should get an indication in your logs when making calls that use deprecated controlnet properties.

**Comment by OedoSoldier** [comment, marked as answer]
OK I figured it out, `control_net_input_image` can pass a list as inputs for different models.

**Reply by ljleb** [reply]
For writing scripts, you may want to consider using the external code API instead of custom `control_net_*` properties defined on `p`. See this portion of the wiki instead for external code API docs: https://github.com/Mikubill/sd-webui-controlnet/wiki/API#create-controlnet-arguments

It should increase the stability of your code to use the API. Calling a function with a fixed signature will scale better than setting properties on objects.
